EventVisualisation/DataConverter/include/EventVisualisationDataConverter/MinimalisticTrack.h

```diff
@@ -0,0 +1,120 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    MinimalisticTrack.h
+/// \author  Jeremi Niedziela
+/// \author  Maciej Grochowicz
+///
+
+#ifndef ALICE_O2_EVENTVISUALISATION_BASE_MINIMALISTICTRACK_H
+#define ALICE_O2_EVENTVISUALISATION_BASE_MINIMALISTICTRACK_H
+
+#include "ConversionConstants.h"
+
+#include <TObject.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <cmath>
+
+namespace o2  {
+namespace EventVisualisation {
+
+/// Minimalistic description of particles track
+///
+/// This class is used mainly for visualisation purpose.
+/// It keeps basic information about a track, such as its vertex,
+/// momentum, PID, phi and theta or helix curvature.
+  
+class MinimalisticTrack : public TObject
```

Comment:
Your classes MinimalisticTrack, MinimalisticEvent, etc. derive from TObject. Is there any hard reason for doing so or could we live without?

Reply:
@sawenzel, there is a reason for that - these classes will be used to export events to JSON and XML using root's built-in engines (e.g. TBufferJSON), which can only work on classes deriving from TObject.

Reply:
Thanks. Can you clarify for which precise API a TObject is needed? In many cases, a dictionary or a TClass might be enough for IO. These things can be generated independently of being a TObject.

Reply:
For example TBufferJSON::ConvertToJSON - from documentation: "Converts object, inherited from TObject class, to JSON string".

Reply:
Yes but there are many overloads of this function: 
```
ConvertToJSON (const void *obj, const TClass *cl, Int_t compact=0, const char *member_name=0)
```

Reply:
I am asking because depending on how many tracks, events are being visualized this question can have quite some dramatic  impact on the memory footprint.

Reply:
So you could convert to json using the following code snippet
```
MinimalisticTrack track;
ConvertToJson(&track, TClass::GetClass("MinimalisticTrack"));
```